src/main/java/org/generationcp/middleware/service/impl/study/StudyInstanceServiceImpl.java

```diff
@@ -367,4 +371,15 @@ private void mapGeolocationMetaData(final Geolocation geolocation, final int var
 		}
 	}
 
+	private void deleteExperimentalDesign(final int studyId, final List<Integer> remainingInstances, final int environmentDatasetId) {
+		//Check if Remaining Instances has ExperimentalDesign
+		final List<StudyInstance> instances = this.daoFactory.getDmsProjectDAO().getDatasetInstances(environmentDatasetId, remainingInstances);
```

Comment:
As discussed, this can be a optimized by simply querying if there is any experiment left for the plot dataset.